src/nvim/option.c

```diff
@@ -7867,20 +7872,18 @@ opt_strings_flags (
     int list                       /* when TRUE: accept a list of values */
 )
 {
-  int i;
-  int len;
-  unsigned new_flags = 0;
+  unsigned int new_flags = 0;
 
   while (*val) {
-    for (i = 0;; ++i) {
+    for (unsigned int i = 0;; ++i) {
       if (values[i] == NULL)            /* val not found in values[] */
         return FAIL;
 
-      len = (int)STRLEN(values[i]);
+      size_t len = STRLEN(values[i]);
       if (STRNCMP(values[i], val, len) == 0
           && ((list && val[len] == ',') || val[len] == NUL)) {
         val += len + (val[len] == ',');
-        new_flags |= (1 << i);
+        new_flags |= (1U << i);
```

Comment:
My edge case scenario: Someone calls this function with at least `sizeof(unsigned)*8` entries in `values`. So it could happen that we try to shift more bits than the first operand has (my assumption here is that this is undefined).